src/main.c

```diff
@@ -86,7 +86,7 @@ int main(int argc, char *argv[])
     {
         // default values
         int batch_size = 200;
-        int dims_hidden[8] = {16};
+        int dims_hidden[8] = {16, 16};
```

Comment:
is this legal?

Reply:
Seems so. This will initialize the first to elements to 16 and leave the rest at 0. Before, only the first was initialized, so default was one hidden layer.

Reply:
Ok! you can also see here that the other values are https://godbolt.org/z/oxYvn9ah6 set to 0.

also https://stackoverflow.com/questions/64946230/initializing-an-int-array-with-less-values-then-that-are-declared-will-it-only